service/src/main/kotlin/fi/espoo/evaka/holidayperiod/HolidayPeriodControllerCitizen.kt

```diff
@@ -25,4 +34,33 @@ class HolidayPeriodControllerCitizen(private val accessControl: AccessControl) {
         accessControl.requirePermissionFor(user, Action.Global.READ_HOLIDAY_PERIODS)
         return db.connect { dbc -> dbc.read { it.getHolidayPeriods() } }
     }
+
+    @PostMapping("/holidays")
+    fun postAbsences(
+        db: Database,
+        user: AuthenticatedUser,
+        @RequestBody body: HolidayAbsenceRequest
+    ) {
+        val childIds = body.childHolidays.keys
+        Audit.HolidayAbsenceCreate.log(targetId = childIds.toSet().joinToString())
+        accessControl.requirePermissionFor(user, Action.Global.CREATE_HOLIDAY_ABSENCE)
+        accessControl.requireGuardian(user, childIds)
+
+        db.connect { dbc ->
+            dbc.transaction { tx ->
+                val childToDateRange = body.childHolidays.entries.filter { it.value != null }.map { it.key to it.value!! }
+                val childToDates = childToDateRange.flatMap { (childId, dateRange) ->
+                    dateRange.dates().map { childId to it }
+                }
+
+                tx.clearOldAbsences(childToDates)
+
+                tx.insertAbsences(PersonId(user.id), childToDateRange.map { (childId, dateRange) -> AbsenceInsert(setOf(childId), dateRange, AbsenceType.FREE_ABSENCE) })
+            }
+        }
```

Comment:
The input needs to be validated against the holiday period in database.

Reply:
Added a separate ticket for this